nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/StandardFlowService.java

```diff
@@ -533,7 +533,9 @@ public void load(final DataFlow dataFlow) throws IOException, FlowSerializationE
                 // operating in standalone mode, so load proposed flow and initialize the controller
                 loadFromBytes(dataFlow, true);
                 initializeController();
-                dao.save(controller, true);
```

Comment:
Saving the flow here is important when interacting with a cluster. I don't think this is something that can/should be removed.